app/controllers/posts_controller.rb

```diff
@@ -32,6 +32,13 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    flash.now[:success] = 'Post deleted successfully.'
+    redirect_to root_path
+  end
+
   private
 
   def set_post
```

Comment:
- [ ] Nice work guys, however, there is just one more state to take to get the project approved 😄. Kindly add the `load_and_authorize_resource` method in the post controller which will raise an exception if the user is not able to perform the given action. This method is provided to automatically authorize all actions in a RESTful style resource controller. It will use a before action to load the resource into an instance variable and authorize it for every action.

```ruby
class PostsController < ApplicationController
  load_and_authorize_resource

  def show
    # @post is already loaded and authorized
  end

  def index
    # @posts is already loaded with all posts the user is authorized to read
  end
end